pkg/controllers/certificatestore_controller.go

```diff
@@ -96,6 +97,13 @@ func (r *CertificateStoreReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		}
 		certificatesMap[resource] = certificates
 		logger.Infof("%v certificates fetched for certificate store %v", len(certificates), resource)
+	case "inline":
```

Comment:
There were some suggestion to moving the certificate providers to a [factory](https://github.com/deislabs/ratify/blob/fd06f53869e65a2bcf264207495dc0ccaf25cb1b/pkg/common/oras/authprovider/authproviderfactory.go#L32) pattern similar to the auth provider, do you see value in that ? We we hoping, by registering each provider,  we won't need the cases statements.  The refactoring is not a blocker for this PR,  I just wanted to check how you feel about moving to the factory pattern, we can  to do the refactoring later. :)

Reply:
Makes sense! Yeah, I think it's a good idea. I actually looked initially for an `init()` factory similar to auth providers/verifiers/stores when I put this together and went with the pattern that was already in place